* Define co_boot_params_t for daemons.
Additional patch after "base" for kernel 2.6.12, 2.6.13, 2.6.14

Index: linux-2.6.12-source/include/linux/cooperative.h
===================================================================
--- linux-2.6.12-source.orig/include/linux/cooperative.h
+++ linux-2.6.12-source/include/linux/cooperative.h
@@ -321,6 +321,16 @@
 	unsigned long compiler_abi;
 } __attribute__((packed)) co_info_t;
 
+typedef struct {
+	unsigned long co_core_end;		// co_passage_page->params[0]
+	unsigned long co_memory_size;		// co_passage_page->params[1]
+	void *co_initrd;			// co_passage_page->params[2]
+	unsigned long co_initrd_size;		// co_passage_page->params[3]
+	unsigned long co_cpu_khz;		// co_passage_page->params[4]
+	unsigned long filler[5];		// compatible api: empty 5,6,7,8,9
+	char co_boot_parameters[CO_BOOTPARAM_STRING_LENGTH]; // co_passage_page->params[10]
+} __attribute__((packed)) co_boot_params_t;
+
 #ifndef COLINUX_TRACE
 #define CO_TRACE_STOP
 #define CO_TRACE_CONTINUE
